### PR TITLE
[123] change login form to nextcloud login flow v2

### PR DESCRIPTION
**Pull request by steffenmalisi**
This PR resolves #123 (https://gitlab.com/nkming2/nc-photos/-/issues/123) by using Nextcloud's login flow v2.

I kept the original structure, but I personally find some files too bloated, e.g. app/lib/bloc/app_password_exchange.dart.
I would prefer to split it in three files under a separate folder app/lib/bloc/app_password_exchange/ : 
- app_password_exchange_bloc.dart
- app_password_exchange_state.dart (part of app_password_exchange_bloc)
- app_password_exchange_event.dart (part of app_password_exchange_bloc)

Let me know what you think :-)

### CODE REVIEW

**Comment by nkming2** [comment]
Hi thanks for the PR! I'm a bit busy right now but I'll take a look ASAP, sorry for the delay!

**Comment by nkming2** [comment]
I tried it on my test server and it failed as my server somehow returned HTTP302 when POSTing to login/v2/poll, I'm investigating it.

**Comment by steffenmalisi** [comment]
Hi, have you been successful with testing? You can also try to use plain curl commands for the login flow to test your local instance without using to app to see if it is misconfigured somehow. Please give me a hint if I can help.

**Comment by nkming2** [comment]
Yes, I got it working now. There's a trailing slash in the URL and Nextcloud was not happy with it. :sweat_smile:

File: app/lib/api/api_util.dart [hunk 1]
```diff
@@ -109,37 +112,130 @@ String getFacePreviewUrlRelative(
...
+        message: "Server responded with an error: HTTP ${response.statusCode}");
+  }
+}
+
+/// Retrieve App Password after successful initiation of login with Nextclouds login flow v2: https://docs.nextcloud.com/server/latest/developer_manual/client_apis/LoginFlow/index.html#login-flow-v2
+Future<AppPasswordResponse> _getAppPassword(
+    InitiateLoginPollOptions options) async {
+  Uri baseUrl;
+  if (options.endpoint.scheme == "http") {
+    baseUrl = Uri.http(options.endpoint.authority);
```
Comment (nkming2) [inline app/lib/api/api_util.dart:140]: It seems that the API was only changed in flutter 3.3 and now the second param is optional (it wasn't in 3.0 I'm using). I'll make the upgrade to 3.3 then.

Comment (nkming2) [reply app/lib/api/api_util.dart:140]: done

**Comment by nkming2** [comment]
Thanks for your contribution! Sorry for the looooong delay but this is now merged. The next update will include this PR and should be ready before the end of month. :tada:

**Comment by steffenmalisi** [comment]
Thanks, @nkming2 for your review and the merge. And, no worries about the delay. Most of us are doing this kind of work in our spare time. So absolutely understandable.

**Pull request merged**